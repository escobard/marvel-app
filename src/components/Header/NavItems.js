import React from "react"
import { MenuItem, RaisedButton } from "material-ui"
import RemoveRedEye from 'material-ui/svg-icons/image/remove-red-eye';
import { Link } from "react-router-dom"

// more on menu icons here: http://www.material-ui.com/#/components/menu
// icons for each menu here: https://material.io/icons/
// explore adding icons later, like so:
/* 
	<Link to={link} key={index} className="links">
		<MenuItem leftIcon={<RemoveRedEye />}>{item}</MenuItem>
	</Link>
*/
const NavItems = (links)  =>{
		return links.map((item, index) => {
			let link = item === "home" ? "/" : "/" + item

			return (
				<Link to={link} key={index} className="links">
					<MenuItem>{item}</MenuItem>
				</Link>
			)
		})
	}
export default NavItems