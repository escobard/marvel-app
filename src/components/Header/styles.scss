@import "../../styles/variables";
@import "../../styles/mixins";

nav.navbar {
	padding: 0;
	margin-bottom: $space-xlrg;
	.appbar {
		background-color: $primary !important;
	}
	h1 {
		text-align: center;
		text-decoration: none;
		color: $white;
		margin-top: $space-small;
	}
	.drawer {
		div {
			&:nth-child(3) {
				top: auto !important;
			}
			&:nth-child(2) {
				background-color: $primary !important;
			}
		}
		// for this app, all material-ui styles will be overwritten with !important
		// this allows for consistent sass utilization to avoid inline styles
		a span {
			color: $white !important;
			text-align: center;
		}
		.navbar-top {
			height: 64px;
			padding-top: $space-med;
			svg {
				margin: auto;
				display: block !important;
				width: $space-xxlrg !important;
				height: $space-xxlrg !important;
				color: $white !important;
			}
		}
	}
	a.links {
		text-transform: capitalize;
		@include nav-links();
	}
}
