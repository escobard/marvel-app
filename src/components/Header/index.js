import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { FlatButton, AppBar, Drawer, RaisedButton } from "material-ui";
import DashIcon from "material-ui/svg-icons/action/dashboard";

import styles from "./styles.scss"
import NavItems from "./NavItems";

// usually all redux state would be handled within a container, but since
// the header component has no container this will be an exception
class Header extends Component {
	constructor(props) {
		super(props);
		this.state = {
			navItems: ["home", "about"]
		};
	}

	handleToggle = () => {
		this.setState({ open: !this.state.open });
	};

	render() {
		let { navItems, open } = this.state;

		return (
			<nav className="navbar navbar-light">
				<AppBar
					className="appbar"
					title={
						<Link to="/" className="navbar-title">
							<h1>InfinityIndex</h1>
						</Link>
					}
					onLeftIconButtonClick={this.handleToggle}
				/>
				<Drawer
					open={open}
					docked={false}
					onRequestChange={open => this.setState({ open })}
					className="drawer"
				>
					<div className="navbar-top">
						<DashIcon />
					</div>
					{NavItems(navItems)}
				</Drawer>
			</nav>
		);
	}
}

export default connect(null)(Header);
