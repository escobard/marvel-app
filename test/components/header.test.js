import React, { Component } from "react";
import { Provider } from "react-redux";
import Header from "../../src/components/Header";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";

// this is the proper way to render the connected components with jest!
describe("Header snapshot", () => {
    const initialState = { output: 100 };
    const mockStore = configureStore();
    let store, container;

    beforeEach(() => {
        store = mockStore(initialState);
        container = shallow(<Header store={store} props />).dive();
    });

    it("+++ render the Header component", () => {
        expect(toJson(container)).toMatchSnapshot();
    });
});
