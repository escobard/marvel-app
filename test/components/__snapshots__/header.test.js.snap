// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot +++ render the Header component 1`] = `
<nav
  className="navbar navbar-light"
>
  <AppBar
    className="appbar"
    iconElementRight={
      <a
        href="auth/google"
      >
        <FlatButton
          className="login"
          disabled={false}
          fullWidth={false}
          label="Log in"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={false}
        />
      </a>
    }
    onLeftIconButtonClick={[Function]}
    onRightIconButtonClick={[Function]}
    showMenuIconButton={true}
    title={
      <Link
        replace={false}
        to="/"
      >
        <img
          alt="EngBook Logo"
          className="logo"
          src="logo.jpg"
        />
      </Link>
    }
    zDepth={1}
  />
  <Drawer
    className="drawer"
    disableSwipeToOpen={false}
    docked={false}
    onRequestChange={[Function]}
    open={false}
    openSecondary={false}
    swipeAreaWidth={30}
    width={null}
    zDepth={2}
  >
    <div
      className="navbar-top"
    >
      <ActionDashboard />
    </div>
    <Link
      className="links"
      key="0"
      replace={false}
      to="/"
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
      >
        home
      </MenuItem>
    </Link>
    <Link
      className="links"
      key="1"
      replace={false}
      to="/dashboard"
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
      >
        dashboard
      </MenuItem>
    </Link>
    <Link
      className="links"
      key="2"
      replace={false}
      to="/addquestion"
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
      >
        addquestion
      </MenuItem>
    </Link>
  </Drawer>
</nav>
`;
